### ISSUE TITLE
Hi. I have a question.

### ISSUE THREAD

**Issue by Heloworkdl** [antocad/FocusOnDepth#1]
Hi. I am going to run (output) the 'model.p' file trained with focusondepth.
What should I do?

I proceeded with the approach of simply modifying the config file, but an error such as 'Missing key(s) in state_dict:
' occurs.

**Comment by antocad** [comment]
Hi,
I have tested just before and it was working (normally if you don't change anything in the config file by default it runs).
You shouldn't rename the .p file because currently we have this in the code: 
`path_model = os.path.join(config['General']['path_model'], 'FocusOnDepth_{}.p'.format(config['General']['model_timm']))`
so you need to have this: `models/FocusOnDepth_vit_base_patch16_384.p`. it's important to let config['General']['model_timm'] equals to vit_base_patch16_384 dans ce cas.
I hope this helped you.

**Comment by Heloworkdl** [comment]
thanks.
i have more question.

The depth data of the example dataset of this model consists of "8-bit" depth data.
The run.py depth output data is of course 8-bit.

How can I train with "16-bit" depth data and infer 16-bit depth output in run.py?

+ After learning and inferring with 16-bit depth data, 255 value 8-bit depth data was inferred.

**Comment by younesbelkada** [comment]
Hi,

The model outputs by default a torch.FloatTensor which is by default 32-bit if I am correct. I think that you are observing a 8-bit precision because you are directly looking at the pixel values of the output image (PIL Image). In run.py, if you want to achieve what you have mentioned, I guess that you need then to store the floating points in another format than a PIL Image. You can add that directly in the function ```run``` at ```FOD/predictor.py```

Hope this helps,

**Comment by Heloworkdl** [comment]
Thanks.
I have a question.

1. Is this repo capable of customizing Min & Max Value of Output Depth?

2. How do I try to create depth by customizing the Min or Max Value of Depth in this repo?

**Comment by younesbelkada** [comment]
Hi,

Thank you for your question!
In this current implementation the depths are normalized between 0 and 1. Therefore, we use a Sigmoid activation function in the Depth estimation head to make sure that the output is between 0 and 1.
If you want to customize the range of the depth, you may want to tweak the output of the Depth estimation head in the forward function of it to customize the range of the output from the model. You can check the file at ```FOD/Head.py``` line 33

Hope this helps,
Y

**Comment by younesbelkada** [comment]
closing this issue, if you need anything else feel free to reopen the issue
